# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: c8a45415dc5b
# Full Hash: c8a45415dc5b0b33b7e9d4b9cd8114ac09a05218
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-05-28 21:48:03
# Description:
#   Bug 1706998: Replace QuotaManager::GetRef().MaybeRecordShutdownStep by static QuotaManager::(Safe)MaybeRecordQuotaClientShutdownStep; r=dom-storage-reviewers,janv
#   
#   There are some few unsafe uses of MaybeRecordShutdownStep where the QuotaManager singleton may be  not (yet or anymore) alive.
#   In order to not add many unnecessary null checks, we drop GetRef and MaybeRecordShutdownStep in favor of
#   QuotaManager::MaybeRecordQuotaClientShutdownStep
# ==============================================================================

diff -r cc21fb9be25a -r c8a45415dc5b dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Fri May 28 05:49:26 2021 +0000
+++ b/dom/quota/QuotaManager.h	Fri May 28 06:46:04 2021 +0000
@@ -112,8 +112,6 @@
   // Returns a non-owning reference.
   static QuotaManager* Get();
 
-  static QuotaManager& GetRef();
-
   // Returns true if we've begun the shutdown process.
   static bool IsShuttingDown();
 
@@ -371,8 +369,20 @@
   void NotifyStoragePressure(uint64_t aUsage);
 
   // Record a quota client shutdown step, if shutting down.
-  void MaybeRecordShutdownStep(Client::Type aClientType,
-                               const nsACString& aStepDescription);
+  // Assumes that the QuotaManager singleton is alive.
+  static void MaybeRecordQuotaClientShutdownStep(
+      const Client::Type aClientType, const nsACString& aStepDescription) {
+    // Callable on any thread.
+
+    MOZ_DIAGNOSTIC_ASSERT(QuotaManager::Get());
+    QuotaManager::Get()->MaybeRecordShutdownStep(Some(aClientType),
+                                                 aStepDescription);
+  }
+
+  // Record a quota client shutdown step, if shutting down.
+  // Checks if the QuotaManager singleton is alive.
+  static void SafeMaybeRecordQuotaClientShutdownStep(
+      Client::Type aClientType, const nsACString& aStepDescription);
 
   // Record a quota manager shutdown step, if shutting down.
   void MaybeRecordQuotaManagerShutdownStep(const nsACString& aStepDescription);
